storage/src/main/scala/ch/epfl/bluebrain/nexus/storage/auth/AuthorizationError.scala

```diff
@@ -0,0 +1,21 @@
+package ch.epfl.bluebrain.nexus.storage.auth
+
+import ch.epfl.bluebrain.nexus.delta.kernel.jwt.TokenRejection
+
+sealed abstract class AuthorizationError(message: String) extends Exception with Product with Serializable {
+  override def fillInStackTrace(): AuthorizationError = this
+  override def getMessage: String                     = message
+}
```

Comment:
I think it makes more sense to say `extends Exception(message)` rather than override? 

Actually, there are more things usable from the Exception constructor:
```
Exception(String message, Throwable cause, boolean enableSuppression,boolean writableStackTrace)
```

Setting `writableStackTrace` to false does the equivalent of what you're writing here

Given that this is likely a pattern we are going to repeat a lot, I would suggest making some sort of `DeltaError` that you could extend instead, which could capture this boilerplate and hide it

Reply:
Of course it's also fine to do this another day, it's not a big deal

Reply:
Done for the first part, for the second, I don't remember if there was a small difference between the two approachs.

But we can have a dedicated task to review this